### PR TITLE
Release 0.21.0

### PR DESCRIPTION
**Pull request by naree**
## What this does?
Create a new release of mu-scala, version 0.21.0

## Checklist

- [x] Reviewed the diff to look for typos, println and format errors.
- [x] Updated the docs accordingly.

### CODE REVIEW

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.21.0-SNAPSHOT"
+version in ThisBuild := "0.21.0"
```
Comment (juanpedromoreno) [inline version.sbt]: I guess we can wait for:

* https://github.com/higherkindness/mu-scala/pull/772
* https://github.com/higherkindness/mu-scala/pull/770
* https://github.com/higherkindness/mu-scala/pull/768

Before releasing the next version?

Comment (fedefernandez) [reply version.sbt]: I'd say yes

Comment (naree) [reply version.sbt]: Yes, I would like to get the mu sbt plugin change out asap as the documentation has already been updated.

Comment (juanpedromoreno) [reply version.sbt]: I've just rolled back to the latest release version so we are not in a rush ;)

Comment (naree) [reply version.sbt]: Cool, will close this PR.

**Pull request closed without merging**